packages/orchestrator-ui-components/src/api/index.ts

```diff
@@ -160,6 +162,48 @@ export class ApiClient extends ApiClientInterface {
                 prefixlen,
         );
     };
+    getFreePortsByNodeSubscriptionIdAndSpeed = (
+        nodeSubscriptionId: string,
+        interfaceSpeed: number,
+        mode: string,
+    ): Promise<ImsPort[]> => {
+        return this.fetchJson(
+            `surf/ims/free_ports/${nodeSubscriptionId}/${interfaceSpeed}/${mode}`,
+        );
+    };
+
+    // legacy : for imsPort selector
+    subscriptions = (
+        tagList: string[] = [],
+        statusList: string[] = [],
+        productList: string[] = [],
+    ): Promise<NodeSubscription[]> => {
+        const statusFilter = `statuses,${encodeURIComponent(
+            statusList.join('-'),
+        )}`;
+        const tagsFilter = `tags,${encodeURIComponent(tagList.join('-'))}`;
+        const productsFilter = `products,${encodeURIComponent(
+            productList.join('-'),
+        )}`;
+
+        const params = new URLSearchParams();
+        const filters = [];
+        if (tagList.length) filters.push(tagsFilter);
```

Comment:
This is probably legacy but is this intended behaviour? This only pushes to the array if the array is > 0

Reply:
Function arg:

`tagList: string[] = [],`

Complete logic, copied from V1 indeed:


```
        const statusFilter = `statuses,${encodeURIComponent(
            statusList.join('-'),
        )}`;
        const tagsFilter = `tags,${encodeURIComponent(tagList.join('-'))}`;
        const productsFilter = `products,${encodeURIComponent(
            productList.join('-'),
        )}`;

        const params = new URLSearchParams();
        const filters = [];
        if (tagList.length) filters.push(tagsFilter);
        if (statusList.length) filters.push(statusFilter);
        if (productList.length) filters.push(productsFilter);

        if (filters.length) params.set('filter', filters.join(','));
```